### DISCUSSION TITLE
Partial order precedences and extras

### DISCUSSION THREAD

**Discussion by susliko** [tree-sitter/tree-sitter#1598] [Q&A] [answered]
[This](https://github.com/tree-sitter/tree-sitter/pull/939) PR introduced an awesome feature of partial order precedences.
But it doesn't seem to work when distinguishing tokens defined in the `extras` section.
Here is a [repo](https://github.com/susliko/tree-sitter-relprecstest/blob/master/grammar.js) with a minimal example.

I'm describing a grammar with both single-line (`\* foo`) and block (`(* foo *)`) comments.
The problem arises with single-line comments, nested into block comments:
```
(* \* *)
```
Specifying higher[ total precedence](https://github.com/susliko/tree-sitter-relprecstest/blob/master/grammar.js#L3-L4) for block comments (e.g. 1 vs 0) helps:
```
(source_file [0, 0] - [1, 0]
  (block_comment [0, 0] - [0, 8]
    (block_comment_text [0, 2] - [0, 6])))
```
But [partial precedences](https://github.com/susliko/tree-sitter-relprecstest/blob/master/grammar.js#L3-L4) fail:
```
(source_file [0, 0] - [1, 0]
  (block_comment [0, 0] - [1, 0]
    (comment [0, 2] - [0, 8])
    (block_comment_text [0, 8] - [1, 0])))
test    0 ms    (MISSING "*)" [1, 0] - [1, 0])
```

Am I misunderstanding the concept of partial precedences or this is an issue?

**Comment by ahelwer** [comment, marked as answer]
It might actually be that partial precedence isn't implemented for lexical precedence, just parser precedence.

**Reply by maxbrunsfeld** [reply]
Yes, that’s right. Sorry about that @susliko , it wasn’t made clear in the PR.
